helpfunctions/utils.py

```diff
@@ -270,7 +277,10 @@ def getMessageState(message):
             return (0, 0)
 
         # Is leaderboard and now find the page of it.
-        pageTopRank = int(str(message.content)[6:10])
+        leaderboard_search_pattern = re.compile("```as\n *[0-9]+\.")
+        leaderboard_search_match = leaderboard_search_pattern.search(str(message.content))
+        leaderboard_search_result = "" if not leaderboard_search_match else leaderboard_search_match.group(0)
+        pageTopRank = int(str(leaderboard_search_result)[6:-1])

```

Comment:
Extract to own function '_get_leaderboard_page(leaderboard_string: str) -> int' in `Utils`

Reply:
Exemplify?